src/io/flutter/run/bazelTest/BazelTestFields.java

```diff
@@ -5,118 +5,147 @@
+import com.jetbrains.lang.dart.sdk.DartSdk;
+import io.flutter.FlutterBundle;
 import io.flutter.bazel.Workspace;
 import io.flutter.bazel.WorkspaceCache;
+import io.flutter.dart.DartPlugin;
 import io.flutter.run.MainFile;
-import io.flutter.run.bazel.BazelFields;
 import io.flutter.run.daemon.RunMode;
+import io.flutter.sdk.FlutterSettingsConfigurable;
+import io.flutter.settings.FlutterSettings;
+import io.flutter.utils.ElementIO;
+import org.jdom.Element;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Map;
+
+import static com.intellij.openapi.util.text.StringUtil.notNullize;
+import static com.intellij.openapi.util.text.StringUtil.nullize;
+
 /**
- * The fields in a Bazel test run configuration, see {@link io.flutter.run.test.TestFields}.
+ * The fields in a Bazel test run configuration.
  */
 public class BazelTestFields {
 
-  @Nullable private String entryFile;
-  @Nullable private String launchScript;
-  @Nullable private String bazelTarget;
+  @Nullable private final String testName;
+  @Nullable private final String entryFile;
+  @Nullable private final String bazelTarget;
 
-  BazelTestFields() {
+  BazelTestFields(@Nullable String testName, @Nullable String entryFile, @Nullable String bazelTarget) {
+    if (testName != null && entryFile == null) {
+      throw new IllegalArgumentException("testName must be specified with an entryFile");
+    }
+    this.testName = testName;
+    this.entryFile = entryFile;
+    this.bazelTarget = bazelTarget;
   }
 
   /**
    * Copy constructor
    */
-  private BazelTestFields(@NotNull final BazelTestFields original) {
-    entryFile = original.entryFile;
-    launchScript = original.launchScript;
-    bazelTarget = original.bazelTarget;
+  BazelTestFields(@NotNull BazelTestFields template) {
+    this(template.testName, template.entryFile, template.bazelTarget);
   }
 
   /**
-   * Create non-template from template.
+   * Returns whether the new test bazel runner is enabled, and if it's available.
    */
-  private BazelTestFields(@NotNull final BazelTestFields template, @NotNull final Workspace workspace) {
-    this(template);
-    if (StringUtil.isEmptyOrSpaces(launchScript)) {
-      launchScript = workspace.getLaunchScript();
-      if (launchScript != null && !launchScript.startsWith("/")) {
-        launchScript = workspace.getRoot().getPath() + "/" + launchScript;
-      }
+  private boolean useNewBazelTestRunner(Project project) {
+    // Check that the new test runner is available.
+    final Workspace workspace = getWorkspace(project);
+    final FlutterSettings settings = FlutterSettings.getInstance();
+    return settings == null ? useNewBazelTestRunnerOverride : settings.useNewBazelTestRunner(project);
+  }
+
+  // The value to use for the bazel test runner setting if no FlutterSettings are available.
+  // TODO: set up a FlutterSettings implementation that we can use in tests.
```

Comment:
We should use `TODO(username):` here, so people will know who to ask for context.